src/Tizen.Network.Bluetooth/Tizen.Network.Bluetooth/BluetoothGatt.cs

```diff
@@ -257,23 +257,160 @@ protected virtual void Dispose(bool disposing)
+                        _taskForDisconnection.SetException(BluetoothErrorFactory.CreateBluetoothException(e.Result));
+                    }
+                    _taskForDisconnection = null;
+                }
+
+                GattConnectionStateChanged?.Invoke(this, e);
+            }
+        }
+
+        private static event EventHandler<GattConnectionStateChangedEventArgs> StaticConnectionStateChanged
+        {
+            add
+            {
+                if (_connectionStateChanged == null)
+                {
+                    RegisterConnectionStateChangedEvent();
+                }
+                _connectionStateChanged += value;
+            }
+            remove
+            {
+                _connectionStateChanged -= value;
+                if (_connectionStateChanged == null)
+                {
+                    UnregisterConnectionStateChangedEvent();
+                }
+            }
+        }
+
+        private static void RegisterConnectionStateChangedEvent()
+        {
+            Interop.Bluetooth.GattConnectionStateChangedCallBack _connectionStateChangeCallback = (int result, bool connected, string remoteDeviceAddress, IntPtr userData) =>
+            {
+                Log.Info(Globals.LogTag, "Setting gatt connection state changed callback");
+                GattConnectionStateChangedEventArgs e = new GattConnectionStateChangedEventArgs(result, connected, remoteDeviceAddress);
+                _connectionStateChanged?.Invoke(null, e);
+            };
+
+            int ret = Interop.Bluetooth.SetGattConnectionStateChangedCallback(_connectionStateChangeCallback, IntPtr.Zero);
+            if (ret != (int)BluetoothError.None)
+            {
+                Log.Error(Globals.LogTag, "Failed to set gatt connection state changed callback, Error - " + (BluetoothError)ret);
+                BluetoothErrorFactory.ThrowBluetoothException(ret);
+            }
+        }
+
+        private static void UnregisterConnectionStateChangedEvent()
+        {
+            int ret = Interop.Bluetooth.UnsetGattConnectionStateChangedCallback();
+            if (ret != (int)BluetoothError.None)
+            {
+                Log.Error(Globals.LogTag, "Failed to unset gatt connection state changed callback, Error - " + (BluetoothError)ret);
+                BluetoothErrorFactory.ThrowBluetoothException(ret);
+            }
+        }
+
+        /// <summary>
+        /// Connects to the Bluetooth GATT client asynchronously.
+        /// </summary>
+        /// <returns>The BluetoothGattClient instance.</returns>
+        /// <feature>http://tizen.org/feature/network.bluetooth.le.gatt.client</feature>
+        /// <exception cref="NotSupportedException">Thrown when the BT/BTLE is not supported.</exception>
+        /// <exception cref="InvalidOperationException">Thrown when the create GATT client fails.</exception>
+        /// <since_tizen> 6 </since_tizen>
+        public Task ConnectAsync(bool autoConnect)
```

Comment:
How about return bool to indicate result?
```suggestion
        public Task<bool> ConnectAsync(bool autoConnect)
```